src/ol/source/Raster.js

```diff
@@ -589,6 +592,17 @@ class RasterSource extends ImageSource {
       this.layers_[i].addEventListener(EventType.CHANGE, changed);
     }
 
+    if (options.resolutions !== null) {
+      for (let i = 0, ii = this.layers_.length; i < ii; ++i) {
+        const source = this.layers_[i].getSource();
+        const resolutions = source.getResolutions();
```

Comment:
This returns the original tilegrid resolutions, but the source could be rendered with a difference grid (e.g. vector tiles or reprojected raster tiles) whose resolutions would be more appropriate here.  Maybe `source.getResolutions()` should take an optional projection to handle that?

Reply:
Good suggestion, implemented. The rendering tests now also work without the previous change.